src/gt4py/next/iterator/transforms/infer_domain.py

```diff
@@ -423,5 +476,10 @@ def infer_program(
         function_definitions=program.function_definitions,
         params=program.params,
         declarations=program.declarations,
-        body=[_infer_stmt(stmt, offset_provider, symbolic_domain_sizes) for stmt in program.body],
+        body=[
+            _infer_stmt(
+                stmt, offset_provider, symbolic_domain_sizes, [param.id for param in program.params]
```

Comment:
```suggestion
                stmt, offset_provider, symbolic_domain_sizes, allowed_unknown_domains=[param.id for param in program.params]
```
for readability

Reply:
I introduced a typed dict `InferenceOptions` that handles forwarding the options in all places where they are not used.